### PR TITLE
Add tox.ini and configure coverage and flake runs

### PR DESCRIPTION
**Pull request by icywang86rui**
* Add more unit tests
* Configure unit tests to run with both py2 and py3
* Add flake checks
* Fix broken integ tests

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+[tox]
+envlist = py27,py36,flake8
+skip_missing_interpreters = False
+
+[travis]
+python =
+  3.6: py36, flake8
+
+[flake8]
+max-line-length = 120
```
Comment (laurenyu) [inline tox.ini]: we may or may not be moving to 100 - I'll chat with you offline

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+[tox]
+envlist = py27,py36,flake8
+skip_missing_interpreters = False
+
+[travis]
+python =
+  3.6: py36, flake8
+
+[flake8]
```
Comment (laurenyu) [inline tox.ini]: should we also include stuff for the import statements? e.g. https://github.com/aws/sagemaker-containers/blob/master/.flake8

Comment (icywang86rui) [reply tox.ini]: I will do that.

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+passenv =
+    TEAMCITY_VERSION
+    AWS_ACCESS_KEY_ID
+    AWS_SECRET_ACCESS_KEY
+    AWS_SESSION_TOKEN
+# {posargs} can be passed in by additional arguments specified when invoking tox.
+# Can be used to specify which tests to run, e.g.: tox -- -s
+commands =
+    coverage run -m pytest {posargs}
+    {env:IGNORE_COVERAGE:} coverage report --fail-under=90 --include *sagemaker_tensorflow_container* --omit */tensorflow/tensorflow_serving/*
```
Comment (mvsusp) [inline tox.ini]: You can separate code coverage between python 2 and python 3 following this example here https://github.com/aws/sagemaker-containers/blob/master/.coveragerc_py27

https://github.com/aws/sagemaker-containers/blob/master/tox.ini#L32

This file here configures codecov https://github.com/aws/sagemaker-containers/blob/master/codecov.yml

**Pull request merged**